File: sort/insertion_sort.go
package sort

// insertion sort - asc
func InsertionSort(inArr []int) {
	for i := 1; i < len(inArr); i++ {
		for z := i; z > 0 && inArr[z] < inArr[z-1]; z-- {
			inArr[z], inArr[z-1] = inArr[z-1], inArr[z]
		}
	}
}

File: sort/insertion_sort.cpp
// +build ignore
// from golang

#include "insertion_sort.h"

using namespace std;

// insertion sort - asc
void insertionSort(vector<int> &inArr) {
    for(int i=1; i<int(inArr.size()); ++i) {
        for(int z=i; z>0 && inArr[z] < inArr[z-1]; --z) {
            int tmp = inArr[z];
            inArr[z] = inArr[z-1];
            inArr[z-1] = tmp;
        }
    }
}